### DISCUSSION TITLE
cli App Generation with Default File structuring

### DISCUSSION THREAD

**Discussion by ekinsipahi** [pallets/flask#5128] [Q&A] [answered]
Why flask does not support this? Thru and CLI command sth like : `flask create-app todo-app` it would be much better because this way file structuring messes up everything is into everything for begginner programmers I think flask has alot of potential but we need default file structuring for the flask framework that would be a much better approach I think. for example a main app that controls everything and other inner apps in a folder by CLI command's ?? By the way my favorite framework is Flask for web dev but we definitely need a file structuring provided my the framework. Thanks

**Comment by ThiefMaster** [comment]
As a beginner you can learn a lot from figuring out a structure that suits you....

**Reply by ekinsipahi** [reply]
I'm not actually a beginner but I started learning backend before with flask and it was such a pain to not to mix everything up so my perspective was maybe a different point of view to light up an idea of how file structuring works with flask.

**Reply by ThiefMaster** [reply]
When I used Flask the first time I also was pretty much a beginner (not even super experienced with Python). Was my app structure perfect? Nope. Did it work? Yes, Does it still work? Yes. Would i structure things differently? Absolutely. Is my structure a problem except when it comes to my personal opinion on what's "nice" or "clean"? Nope.

**Comment by davidism** [comment, marked as answer]
Flask intentionally requires no structure beyond "single Python module", and no specific libraries for other services. Check out [cookiecutter](https://cookiecutter.readthedocs.io), it's already a great framework for creating and using project templates. If you want an opinionated starting point, there are already multiple templates for Flask projects.
